### PR TITLE
Add ability to create coupons for credit seats.

### PR DESCRIPTION
**Pull request by vkaracic**
This PR contains changes that allow the creation of coupons for credit seats.
screenshot from 2016-09-21 14-45-42
screenshot from 2016-09-21 14-45-47

Also additional custom jasmine matches are added and test updated accordingly.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=947

File: ecommerce/extensions/api/v2/views/coupons.py [hunk 1]
```diff
@@ -106,6 +106,11 @@ def create(self, request, *args, **kwargs):
             invoice_data = self.create_update_data_dict(data=request.data, fields=Invoice.UPDATEABLE_INVOICE_FIELDS)
 
             if course_seat_types:
+                if len(course_seat_types) > 1 and 'credit' in course_seat_types:
```
Comment (marjev) [inline ecommerce/extensions/api/v2/views/coupons.py]: Add this as a custom validator for course_seat_types Range field. If it fails there, ValidationError will be raised instead.

**Comment by vkaracic** [comment]
@mjfrey please review.

**Comment by vkaracic** [comment]
Comments addressed.

**Review by marjev — APPROVED** [review]

File: ecommerce/extensions/offer/tests/test_models.py [hunk 2]
```diff
@@ -122,6 +124,20 @@ def test_query_range_all_products(self):
         self.range.course_seat_types = 'verified'
         self.assertEqual(len(self.range.all_products()), 0)
 
+    def test_credit_seat_type_validator(self):
+        """
+        Verify the validator raises error for combination of credit and another seat type.
+        """
+        ver_pro_types = 'verified,professional'
+        credit_type = 'credit'
+        invalid_type = 'credit,verified'
```
Comment (iivic) [inline ecommerce/extensions/offer/tests/test_models.py]: Are these variables used only once? If so why not use the strings directly?

**Comment by vkaracic** [comment]
Comments addressed.

**Review by iivic — APPROVED** [review]

**Comment by mjfrey** [comment]
👍

**Pull request merged**